src/core/mac/mac.cpp

```diff
@@ -1375,7 +1377,48 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
         break;
 #endif
 
+        // count Oob frames based on type
+
     case kOperationTransmitOutOfBandFrame:
+        switch (aFrame.GetType())
```

Comment:
I won't recommend parsing the OOB frame to update the counters (e.g., determine if is if MAC command etc).
Would suggest just incrementing the `mTxOther` to keep it simple.

The intention behind the OOB frame was to send a custom frame (application/vensor specific) frame  which may not even follow full 802.15.4 frame/header format and/or may be encrypted differently.

In longer term I think it would be good to remove the whole OOB feature (it was intended to solve a problem for a legacy networking solution). If such a use is needed in future, I would recommend handling it at platform layer and radio driver level (instead of having an API which allows a frame to be injected through OT code).

Reply:
@abtink Thanks for the suggestion, the simple way is increasing mTxOther.  In the discussion for issue[#4459](https://github.com/openthread/openthread/issues/4459) ,  the recommended solution is counting Oob frames based on the frame type.  Thoughts? @bukepo

Reply:
Agree that counting OoB into mTxOther is a simple solution.
@bukepo thoughts?

Reply:
In #4459
> Describe alternatives you've considered
> count these frames as other frames because there's other rx frame counter.

If `rx` is a typo for `tx`, the alternative should be the simple solution proposed here? @bukepo @FiveDimensions